### ISSUE TITLE
Difficulty putting a footnote after a curvy quote

### ISSUE THREAD

**Issue by jinwoo** [asciidoctor/asciidoctor#3945]
I have to use a trick similar to https://github.com/asciidoctor/asciidoctor/issues/3467 for the case where a footnote comes after a curvy quote. For example, when you write

```adoc
He said "`Hello, world.`"footnote:[This is a footnote.]
```

, the closing quote is not curved because it isn't followed by a delimiter. It is rendered `` `" `` instead. My workaround is

```adoc
He said "`Hello, world.`"{wj}footnote:[This is a footnote.]
```

It'd be great if asciidoctor could treat macros like `footnote:[]` as a word delimiter and handle the curvy quote correctly.

**Comment by mojavelinux** [comment]
I agree that the adjacency of footnotes in AsciiDoc is a pain point that needs to be addressed. This is something I want the asciidoc-lang-dev (the spec) to look into. Until then, there's not much we can do about it in Asciidoctor as we aren't making changes to the language until that project is up and running. Until then, I'll mark this for consideration by the spec project / committee.

**Comment by jinwoo** [comment]
A tangential issue related to curvy quotes followed by an em dash.

```adoc
"`hello`"--world
```

doesn't render `--` as an em dash. Just two dashes instead. I have to put spaces around `--`.

```adoc
"`hello`" -- world
```

But this puts thin spaces around the em dash. Maybe the em dash should always put word joiners around it instead of thin spaces?

**Comment by mojavelinux** [comment]
> A tangential issue related to curvy quotes followed by an em dash.

Please don't use the same issue to discuss different aspects of the syntax.

**Comment by mojavelinux** [comment]
> This is something I want the asciidoc-lang-dev (the spec) to look into.

To be clear, what I have in mind is that you put a space in front of the footnote macro, and that space will be dropped by the parser (treated as insignificant)

```
He said "`Hello, world.`" footnote:[This is a footnote.]
```

We'll only then have to address the situation when the intention is to add a space in front of the footnote (which is rare). We might allow that space to be escaped to communicate that intent. The backslash would indicate that the space should be preserved.

```
He said "`Hello, world.`"\ footnote:[This is a footnote.]
```

**Comment by jinwoo** [comment]
Sorry for mixing two unrelated issues. I created https://github.com/asciidoctor/asciidoctor/issues/3946 instead.

Making the space always dropped sounds great to me. That will also make the source text document more readable. Thank you very much for the quick response!

**Comment by mojavelinux** [comment]
> Making the space always dropped sounds great to me. That will also make the source text document more readable.

Excellent to hear! It really does feel like the right thing to do.

As soon as the asciidoc-lang-dev project is running, I'm open to prototyping this in Asciidoctor ASAP. It would certainly help us gather feedback so that we can make it part of the language.
